api/src/main/java/org/pmiops/workbench/cohorts/CohortService.java

```diff
@@ -68,11 +100,31 @@ public CohortService(
         .build();
   }
 
-  public TerraJob submitGenomicsCohortExtractionJob(String workspaceNamespace, String workspaceName)
+  private List<String> getPersonIds(Long cohortId) {
+    String cohortDefinition = cohortDao.findOne(cohortId).getCriteria();
+
+    final SearchRequest searchRequest = new Gson().fromJson(cohortDefinition, SearchRequest.class);
+
+    final QueryJobConfiguration participantIdQuery =
+            cohortQueryBuilder.buildParticipantIdQuery(new ParticipantCriteria(searchRequest));
+
+    return Streams.stream(bigQueryService.executeQuery(bigQueryService.filterBigQueryConfig(participantIdQuery))
+            .getValues())
+            .map(personId -> personId.get(0).getValue().toString())
+            .collect(Collectors.toList());
+  }
+
+  public TerraJob submitGenomicsCohortExtractionJob(DbWorkspace workspace, Long cohortId)
       throws ApiException {
     WorkbenchConfig.WgsCohortExtractionConfig cohortExtractionConfig =
         workbenchConfigProvider.get().wgsCohortExtraction;
 
+    FirecloudWorkspace fcWorkspace = fireCloudService.getWorkspace(workspace).get().getWorkspace();
+    String extractionUuid = UUID.randomUUID().toString();
+    String filename = extractionUuid + "_person_ids.txt";
+    Blob personIdsFile = cloudStorageService.writeFile(
```

Comment:
I was thinking we may have to spin this off into a separate thread but it seems fast enough for now. I was able to write 100k sample names without a hitch.

I think it may still be worth switching over but I was thinking we can move it into a new ticket.